extensions/jaxb/runtime/src/main/java/io/quarkus/jaxb/runtime/JaxbContextConfigRecorder.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.jaxb.runtime;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class JaxbContextConfigRecorder {
+    private volatile static Set<String> classesToBeBound = new HashSet<>();
+
+    public void addClassesToBeBound(Collection<String> additionalClassesToBeBound) {
+        this.classesToBeBound.addAll(additionalClassesToBeBound);
+    }
+
+    public static String[] getClassesToBeBound() {
+        return classesToBeBound.toArray(new String[classesToBeBound.size()]);
```

Comment:
```suggestion
        return classesToBeBound.toArray(new String[0]);
```

Reply:
I think you can have a method here returning Class[] since that's what's going to be used when invoking `JAXBcontext.newInstance`

Reply:
I prefer doing the mapping from string to class in the `JAXBContext.newInstance` method.